services/workflows-service/src/collection-flow/collection-flow.service.ts

```diff
@@ -229,6 +235,51 @@ export class CollectionFlowService {
     return flowData;
   }
 
+  private async __persistFileUrlsToDocuments(
+    documents: TDocument[] = [],
+    projectIds: TProjectIds,
+  ): Promise<TDocument[]> {
+    const fileEntities = (
+      await Promise.all(
```

Comment:
In case one of the files has failed, what is the expected behavior ?

In case we would like to be more resilient, i can suggest to use `Promise.allSettled`

`Promise.allSettled` will never reject, it will resolve once all Promises in the array have either rejected or resolved

Reply:
Let's add to tech debt a task or a chance to revisit instances of `Promise.all` and discuss how we should handle them. Handling will be in a separate PR.

Reply:
Currently if any of them fail then 500 excepted,edge case in current scope.